Source/Core/Core/ExecutionHandling/ContextBuilderFactory.cs

```diff
@@ -100,9 +103,45 @@ public static void InitializeBuilders()
         /// </summary>
         internal static void AddBuilderFactory(Func<IIocContainer, IDataStore, IBuilder> builderFactory) => BuilderFactories.Add(builderFactory);
 
-        /// <summary>
-        /// Eventually, this method will unload the separate app domains set up during <c>Initialize</c>.
-        /// </summary>
-        public static void Cleanup() {}
+        /// <summary>Cleanup by disposing registered disposables.</summary>
+        public static void Cleanup() => DisposablesForCleanup.Clear();
+
+        /// <summary>Register to be disposed at cleanup.</summary>
+        public static void AddForCleanup(IDisposable disposable) => DisposablesForCleanup.Add(disposable);
```

Comment:
In addition to these, would it make sense to add a method that sets it up to dispose the created IoC container itself (if it supports it?). The IoC containers I worked with usually implement `IDisposable` which cleans up any managed `IDisposable` instances (might need to make the `IIocContainer` implement `IDisposable` as well).

Reply:
Good point. We need to consider that in another PR.